src/auditlog/models.py

```diff
@@ -232,6 +237,73 @@ def changes_str(self, colon=': ', arrow=smart_text(' \u2192 '), separator='; '):
 
         return separator.join(substrings)
 
+    @property
+    def changes_display_dict(self):
+        """
+        :return: The changes recorded in this log entry intended for display to users as a dictionary object.
+        """
+        # Get the model
+        model = self.content_type.model_class()
+        changes_display_dict = {}
+        # grab the changes_dict and iterate through
+        for field_name, values in iteritems(self.changes_dict):
+            # try to get the field attribute on the model
+            try:
+                field = model._meta.get_field(field_name)
+            except FieldDoesNotExist:
+                # use the reverse field mapping to get the field
+                from auditlog.registry import auditlog
+                model_fields = auditlog.get_model_fields(model._meta.model)
+                reversed_field = model_fields['reverse_mapping_fields'].get(field_name)
+                field = model._meta.get_field(reversed_field)
+
+            values_display = []
+            # handle choices fields and Postgres ArrayField to get human readable version
+            if field.choices or hasattr(field, 'base_field') and getattr(field.base_field, 'choices', False):
+                choices_dict = dict(field.choices or field.base_field.choices)
+                for value in values:
+                    try:
+                        value = ast.literal_eval(value)
+                        if type(value) is [].__class__:
+                            values_display.append(', '.join([choices_dict.get(val, 'None') for val in value]))
+                        else:
+                            values_display.append(choices_dict.get(value, 'None'))
+                    except ValueError:
+                        values_display.append(choices_dict.get(value, 'None'))
+                    except:
+                        values_display.append(choices_dict.get(value, 'None'))
+            else:
+                for value in values:
+                    # handle case where field is a datetime type
+                    if "DateTime" in field.get_internal_type():
+                        try:
+                            value = parser.parse(value)
+                            value = value.strftime("%b %d, %Y %I:%M %p")
```

Comment:
I would rather see use of Django's l10n stuff for this, so that the format adapts to different settings.

Reply:
Alright, modified to now rely on `USE_L10N` first and fallback on `settings.DATETIME_FORMAT`, `settings.DATE_FORMAT`, and `settings.TIME_FORMAT` which have defaults and can be set if the user desires.